pkg/controller/cloud/nodecontroller.go

```diff
@@ -155,3 +270,120 @@ func (cnc *CloudNodeController) Run() {
 		}, cnc.nodeMonitorPeriod, wait.NeverStop)
 	}()
 }
+
+func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
+	node := obj.(*v1.Node)
+	instances, ok := cnc.cloud.Instances()
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("cloudprovider does not support instances"))
+		return
+	}
+
+	// This initializes nodes with cloud info
+	// Only initializes nodes that were created with the "ExternalCloudProvider" taint
+	taints := node.Spec.Taints
+
+	var cloudTaint *v1.Taint
+	for _, taint := range taints {
+		if taint.Key == CloudTaintKey {
+			cloudTaint = &taint
+		}
+	}
+
+	if cloudTaint == nil {
+		glog.V(2).Infof("This node %s is registered without the cloud taint. Will not process.", node.Name)
+		return
+	}
+
+	err := clientretry.RetryOnConflict(UpdateNodeSpecBackoff, func() error {
+		curNode, err := cnc.kubeClient.Core().Nodes().Get(node.Name, metav1.GetOptions{})
```

Comment:
The client framework is honestly confusing. There is a CoreV1(), and Core().V1(), what's the diff? and when to use which?